### ISSUE TITLE
Provide default innerContent in eventContent so it can be wrapped

### ISSUE THREAD

**Issue by Gustl22** [fullcalendar/fullcalendar#5927]
Unlike [Tooltip.js](https://fullcalendar.io/docs/v4/event-tooltip-demo) I cannot attach the tooltip to a rendered element. The content must be wrapped inside the [MUI Tooltip](https://material-ui.com/components/tooltips/) before rendering:

```
<Tooltip title="Tooltip">
  <Content>Arrow</Content>
</Tooltip>
```

Whereas `Content` has to be the original `eventContent` dependent on the view (month / list etc.). Then I return the Element via `eventContent`.

References: 
My issue Wordpress implementation: https://github.com/Oberhauser-Dev/gb-fullcalendar/issues/18
See [eventContent hook](https://github.com/Oberhauser-Dev/gb-fullcalendar/blob/088995c6e5c58e67a5ee3d3da7eaec48d9121422/src/GbFullCalendar.js#L82) and [Tooltip workaround](https://github.com/Oberhauser-Dev/gb-fullcalendar/blob/088995c6e5c58e67a5ee3d3da7eaec48d9121422/src/GbFullCalendar.js#L307).

How can I get the appropriate innerContent element in the `eventContent` hook (e.g. via `args`)?

**Comment by acerix** [comment]
I can see how it could be useful to extend the default content, moving to main since it's not specific to react.

It may work better to attach this tooltip from `eventDidMount`.

**Comment by Gustl22** [comment]
Thank you for attending to this issue and moving it to the right place; thought `eventContent` is only usable for React. 

If there's a straight solution with `eventDidMount` I would be happy to use it.
I had some attempts with `eventDidMount` ([here the sandbox example](https://codesandbox.io/s/blissful-cloud-703nx?file=/src/DemoApp.jsx)), which kind of works, but crashes, if react tries to change the DOM itself (e.g. when switching to listView). Also it is a very dirty solution. It would be much cleaner to use the react element directly the `eventContent` method.

**Comment by bahachammakhi** [comment]
I got the same needs here , cant access the original wrapper , went copying code from fullcalendar code to render the same elements ! Hope this gets some improvements soon.

**Comment by setefocos** [comment]
I lost days trying to get the default content... it would be so useful.

**Comment by RyanEwen** [comment]
I am stuck on this as well. I want prepend a tag to my event name, and add some classnames to the wrapper, but I can't do this without nuking the default HTML structure of the event. I have tried to mimic the default event structure but it differs per-view, and probably differs per-event in some cases as well :( 

```
return (
            <div className="fc-event-main-frame">
                <div className="fc-event-time">{info.timeText}</div>
                <div className="fc-event-title-container">
                    <div className="fc-event-title">{eventStatus}{info.event.title}</div>
                </div>
            </div>
        )
```

**Comment by RyanEwen** [comment]
Alright I managed to accomplish what I wanted to do using `eventDidMount` and old-school javascript DOM manipulation.

```
alterEventContent = (info) => {
    // adding a CSS classname
    if (info.event.extendedProps.status = 'rescheduled') {
        info.el.classList.add('rescheduled-event')
    }

    // manipulating the event title
    let titleEl = info.el.getElementsByClassName(info.view.type == 'listWeek' ? 'fc-list-event-title' : 'fc-event-title')[0]

    if (titleEl) {
        // add tooltip
        titleEl.setAttribute('title', titleEl.innerText)
        
        // adding HTML
        let status = this.generateStatusHTML(info.event.extendedProps.status) 
        // eg: '<span class="status-rescheduled">Rescheduled</span>'
        
        titleEl.innerHTML = `${status} ${titleEl.innerHTML}`
    }
}
```

**Comment by philipstratford** [comment]
I agree that this seems like a glaring oversight. I want to conditionally add a simple bit of HTML to my event's content but using `eventContent` gives no way to get the original HTML so that I can just modify it before re-injecting it, and `eventDidMount` provides no way to modify the HTML.

**Comment by philipstratford** [comment]
@RyanEwen I've tried to use your solution but the line 
````
titleEl.innerHTML = `${status} ${titleEl.innerHTML}`
````
 doesn't seem to actually modify the html displayed in the calendar.

**Comment by morninger** [comment]
I'm having a similar issue, working with tooltips.
`eventDidMount` works, but if I edit an event, the event data (duration, text, resource if present etc.) are changed on the calendar but the tooltips keeps showing the old data (I guess it's happening because the event has already been added to the DOM);
`eventContent` seems not able to update tooltips data.
But if I drag the event somewhere else (e.g. I move it to another resource), all the data are updated, tooltips included.

If anybody has an alternative...

**Comment by jamesgour** [comment]
Any progress on this? Anyone know a solution?

**Comment by davidbairdala** [comment]
> Update: turns out it was implemented so that returning true from eventContent will use the default render :) Thanks guys!

Sounds like this info should be in the docs.

**Comment by philipstratford** [comment]
> Update: turns out it was implemented so that returning true from eventContent will use the default render :) Thanks guys!

This is good if you just want to return the event content unmodified, but still doesn't allow for the default HTML to be modified before being returmed. Having innerContent as a parameter within the event handler would be better.

**Comment by SteveSonoa** [comment]
> > Update: turns out it was implemented so that returning true from eventContent will use the default render :) Thanks guys!
> 
> Sounds like this info should be in the docs.

Agreed!
